tests/testthat/test-table_with_settings_ui.R

```diff
@@ -0,0 +1,92 @@
+library(shinytest2)
+
+testthat::test_that("maxi version", {
+  skip_on_cran()
+  skip_on_ci()
+  app <- AppDriver$new(
+    app_tws(),
+    name = "tws",
+    variant = "app_tws_max",
+    )
+  # check standard file naming
+  file_name <- app$get_value(input = "table_with_settings-downbutton-file_name")
+  testthat::expect_equal(
+    sub("_\\d{6}$", "", file_name),
+    paste0("table_", strftime(Sys.Date(), format = "%Y%m%d"))
+  )
+  app$expect_text("table")
+  app$set_window_size(width = 1392, height = 763)
+
+  # save txt (default)
+  app$click("table_with_settings-downbutton-dwnl")
+  app$set_inputs(`table_with_settings-downbutton-dwnl_state` = TRUE)
+  app$set_inputs(`table_with_settings-downbutton-file_name` = "table1")
+  app$expect_download("table_with_settings-downbutton-data_download")
+
+  # save csv:
+  app$set_inputs(`table_with_settings-downbutton-file_format` = ".csv")
+  app$set_inputs(`table_with_settings-downbutton-dwnl_state` = FALSE)
+  app$set_inputs(`table_with_settings-downbutton-file_name` = "table2")
+  app$expect_download("table_with_settings-downbutton-data_download")
+
+  # download using modal, txt first
+  app$click("table_with_settings-expand")
+  app$set_inputs(`table_with_settings-modal_downbutton-lpp` = 70)
+  app$click("table_with_settings-modal_downbutton-dwnl")
+  app$set_inputs(`table_with_settings-modal_downbutton-file_format` = ".txt")
+  app$set_inputs(`table_with_settings-modal_downbutton-pagination_switch` = FALSE)
+  app$click("table_with_settings-modal_downbutton-dwnl")
+  app$set_inputs(`table_with_settings-modal_downbutton-dwnl_state` = TRUE)
+  app$set_inputs(`table_with_settings-modal_downbutton-file_name` = "table3")
+  app$set_inputs(`table_with_settings-modal_downbutton-pagination_switch` = TRUE)
```

Comment:
what's the reason for clicking the button twice and changing pagination_switch twice without any expectations in between?

P.S. this click (`app$click("table_with_settings-expand")`) makes sense than the one from line 21 (`app$click("table_with_settings-downbutton-dwnl")`) because modal is not in html-source from the beginning, but only when clicked.